optuna/integration/sklearn.py

```diff
@@ -69,6 +68,22 @@ def _check_sklearn_availability():
         )
 
 
+def _is_arraylike(x):
+    # type: (Any) -> bool
+    return hasattr(x, '__len__')
+
+
+def _index_param_value(
+    X,  # type: TwoDimArrayLikeType
+    v,  # type: Any
+    indices  # type: OneDimArrayLikeType
+):
+    # type: (...) -> Union[OneDimArrayLikeType, TwoDimArrayLikeType]
+    if not _is_arraylike(v) or _num_samples(v) != _num_samples(X):
+        return v
```

Comment:
https://github.com/scikit-learn/scikit-learn/commit/46001e7cfb163b73a83d5adf73fb454e8a3c64b6#diff-60033c11a662f460e1567effd5faa6f0L949
The original implementation cared for a sparse matrix.
Should we also follow?